### DISCUSSION TITLE
2021-11-17 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#829] [Maintainer discussions]
## About
Each week (usually on Tuesday), we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Progress & Weekly Goals](https://github.com/centerofci/mathesar/discussions/829#discussioncomment-1661436)
- [Welcome Dominykas!](https://github.com/centerofci/mathesar/discussions/829#discussioncomment-1661442)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/829#discussioncomment-1661453)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/829#discussioncomment-1665943)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/829#discussioncomment-1666395)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/829#discussioncomment-1666960)
- [Brent's check-in](https://github.com/centerofci/mathesar/discussions/829#discussioncomment-1669538)
- [Mukesh's check-in](https://github.com/centerofci/mathesar/discussions/829#discussioncomment-1669984)
- [Dominykas' check-in](https://github.com/centerofci/mathesar/discussions/829#discussioncomment-1670085)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/829#discussioncomment-1672293)

## Weekly check-in
@dmos62 @ghislaineguerin @mathemancer @pavish @seancolsen @silentninja please respond to the weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

- How did the last week go? What went well, what could have gone better?
- What are you planning to work on over the next week? Do you anticipate any blockers?
- Anything else you'd like to add?

**Comment by kgodey** [comment]
## Progress & Weekly Goals

We closed milestone **[[06] General Improvements #4](https://github.com/centerofci/mathesar/milestone/31)** (22 issues) and make some progress with subsequent milestones. Nice work, everyone!

Our goals for this week are:

### Design
- Complete the design for #780 
- Complete the foreign key design and finalize the spec:  https://github.com/centerofci/mathesar-wiki/pull/12
- Have a first draft of the viewing views design: #466 

@ghislaineguerin @ppii775 Does that sound good? Anything to add?

### Frontend
- Close some issues in **[[07] Working with Tables](https://github.com/centerofci/mathesar/milestone/8)**
- Get the toast system and modal refactor merged:
  - #821 
  - #828  

@seancolsen @pavish Does that sound good? Anything to add?

### Backend
- Close some issues in **[[08] Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)**
- Figure out API error structure: #560 

@dmos62 @mathemancer @silentninja Does that sound good? Anything to add?

### Product
- Finish remaining Views issues and open them for review: #442

**Reply by pavish** [reply]
> Frontend: Does that sound good? Anything to add?

@kgodey Sounds good to me.

**Reply by seancolsen** [reply]
@kgodey Also sounds good to me

**Reply by mathemancer** [reply]
For back-end, if we have time, I think we should try to look into #592 .

Edit:  other than that, I think the back-end plan makes sense.

**Comment by kgodey** [comment]
## Welcome Dominykas!

I'd like to welcome @dmos62 to the full-time Mathesar team. He'll be working on the backend. @dmos62, perhaps you could tell the community a little more about yourself?

**Reply by dmos62** [reply]
Excited! I got to know many of you already, so I won't bore you with a long bio. `dogs > cats && outdoors > indoors && yoga == true && cold > hot && sea > mountains && isGeek == true`.

**Comment by kgodey** [comment]
## Comms Assignee schedule

FYI, we may adjust the December schedule closer to the holidays once we figure out when people are taking time off.

Upcoming Comms Assignee schedule:
| Week | Person |
|-|-|
| Nov 15 to Nov 19 | Brent |
| Nov 22 to Nov 26 | Kriti |
| Nov 29 to Dec 3 | Pavish |
| Dec 6 to Dec 10 | Mukesh |
| Dec 13 to Dec 17 | Sean |
| Dec 20 to Dec 24 | Dominykas |
| Dec 27 to Dec 31 | Ghislaine |

You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by seancolsen** [comment]
## Sean's check in

How did the last week go? What went well, what could have gone better?

- Things are going well I think. I'm happy to be making some progress on the lower-level components like the toast sytem because I think it's going to speed up the higher-level work.

What are you planning to work on over the next week? Do you anticipate any blockers?

- Modals, toasts, table renaming, and maybe TypeScript strict

Anything else you'd like to add?

- Nope

**Comment by pavish** [comment]
## Pavish's check in

> How did the last week go? What went well, what could have gone better?

Last week went good, I got both the url sync PR and the integ test env PR merged. I'm working on the default value for column PR, which also handles abstracting away type based configuration on the frontend

I also reviewed a number of PRs and discussed & decided on approaches for modals, toasts, dropdown menus etc., I was a bit distracted due to some personal stuff that came up, so there was a delay in responding to discussions and PR reviews.

>  What are you planning to work on over the next week? Do you anticipate any blockers?

* Setting default value for column (includes abstracting away type based configuration, and creation of toggle appearance for Checkbox as per the UX)
* Column renaming
* Adding multi-column unique constraints
   - @ghislaineguerin @kgodey The latest UX for foreign key constraints showed a different UI for the constraints modal. I would like to establish how the modal is supposed to look before starting this issue.

> Anything else you'd like to add?

Nothing at the moment.

**Comment by ghislaineguerin** [comment]
## Ghislaine's Check-In

> How did the last week go? What went well, what could have gone better?

It went well, and I can't think of any specific improvements. The spec creation and review process is becoming more efficient with every iteration.

> What are you planning to work on over the next week? Do you anticipate any blockers?

Close spec for FK Usage. I think a blocker might be the constraints modal changes as pointed out by @pavish. The initial design incorporated a modal inside modal pattern that we wanted to get away from, but we don't have specs for the new one.

> Anything else you'd like to add?

Nothing at the moment.

**Comment by mathemancer** [comment]
## Brent's check-in

> How did the last week go? What went well, what could have gone better?

It went rather poorly.  A number of outside factors kept me from spending large contiguous blocks of time, making coding difficult.  Regardless, I made progress on grouping, and got a number of PRs reviewed and merged, and participated in some discussions.

> What are you planning to work on over the next week? Do you anticipate any blockers?

I've cut down some of the scope on grouping, and plan to get that merged ASAP.  Then, I'll move to some smaller issues around types.

> Anything else you'd like to add?

Nothing at the moment.

**Comment by silentninja** [comment]
## Mukesh's check-in

> How did the last week go? What went well, what could have gone better?

It was okay, was able to sort out the basic structure for implementing display options and make progress on display options for each type. Quite happy to start discussions that would improve the architecture.

Faced some bugs with the framework that delayed things a bit, making the progress a little bit slow.

> What are you planning to work on over the next week? Do you anticipate any blockers?

Planning to work on the timestamp data type and URI filtering options

> Anything else you'd like to add?

Nothing at the moment.

**Comment by dmos62** [comment]
## Dominykas' check-in

> How did the last week go? What went well, what could have gone better?

Everything went great. Excited about my first days on the team. The excitement may have cost some focus, actually.

> What are you planning to work on over the next week? Do you anticipate any blockers?

On implementing filtering for the Number type (#385), which involves laying out the general design for filters. I don't anticipate blockers, but might be on the look out for a second task in case I get blocked.

> Anything else you'd like to add?

Nothing at the moment.

**Reply by kgodey** [reply]
@dmos62 These might be good tasks to look into as a backup:
- #592 
- #280

**Comment by kgodey** [comment]
## Kriti's check in

> How did the last week go? What went well, what could have gone better?

It went pretty well. I'm glad to get a milestone closed and the integration testing environment set up. I also have more clarity on Views (see [wiki page](https://wiki.mathesar.org/en/product/concepts/views)). I would have liked to get the ticket work for Views done, though.

> What are you planning to work on over the next week? Do you anticipate any blockers?

- Views tickets
- #560 
- Project planning
- Presentation for board meeting
- Syncing labels and milestones across all repos, if I have time.
